documentation/The OpenTK Style Guide/main.tex

```diff
@@ -0,0 +1,740 @@
+\end{must}
+
+\begin{must}{Locals must be named correctly}
+All locals (local variables, method parameters, lambda arguments, etc) must be named according to \texttt{camelCase}.
+\end{must}
+
+\begin{must}{Named \texttt{ValueTuple} members must be named correctly}
+Named \texttt{ValueTuple} members are, for all intents and purposes, considered public properties on a class. Therefore, they must be named according to \texttt{camelCase}.
+\end{must}
+
+\section{Declarations}
+\begin{must}{Declarations must be correctly ordered}
+Members in a type declaration must be declared in the following order:
+\begin{itemize}
+\item Public fields
+\item Private fields
+\item Public properties
+\item Private properties
+\item Constructors
+\item Public methods
+\item Private methods
+\item Overridden methods
+\item Interface implementations
+\end{itemize}
+
+Any static declaration must come last in its corresponding section. Within a section, there is no sorting order. Apply common sense.
+\end{must}
+
+\begin{must}{Types must be declared within a namespace}
+Any type declaration must appear inside a namespace.
+\end{must}
+
+\subsection{Methods}
+\begin{should}{Methods should be private}
+Whenever possible, methods should be declared private, or public as part of an internal class.
+\end{should}
+
+In order to reduce the public API surface of the library beyond what is actually maintained, the set of methods exposed to the end user should be as small as possible.
+
+\subsection{Properties}
+\begin{should}{Properties should be auto-properties}
+Whenever possible, a property's accessors should be automatically implemented.
+\end{should}
+
+\begin{must}{Auto-properties must be declared on one line}
+If a property is declared as an auto-property, it must be declared on one line.
+\end{must}
+
+That is, the following example 
+\begin{code}[One line]
+public bool IsCorrect { get; set; }
+\end{code}
+
+is to be preferred over
+
+\begin{code}[Multiple lines]
+public bool IsCorrect 
+{
+    get; 
+    set; 
+}
+\end{code}
+
+\begin{should}{Properties should be expression-bodied}
+Whenever possible, a property should be expression-bodied.
+\end{should}
+
+That is, the following example 
+\begin{code}[One line]
+public int GetSomeValue => _internalValue * 5;
+\end{code}
+
+is to be preferred over
+
+\begin{code}[Multiple lines]
+public bool IsCorrect 
+{
+    get
+    {
+        return _internalValue * 5;
+    }
+}
+\end{code}
+
+
+\section{Documentation}
+\begin{should}{Private members should be documented}
+Whenever possible, the internal API surface should be documented.
+\end{should}
+
+In order to help the next person along, all members should be documented - not just the members exposed to the end user.
+
+\begin{must}{Files must have the correct header}
+Files shall contain a file header, containing the following text.
+\end{must}
+
+\begin{code}[File header]
+// 
+//  <filename>
+//
+//  Copyright (C) <year of creation> OpenTK
```

Comment:
Previous authors cannot be changed or removed, they should be mentioned.

Reply:
Good point. If the file has listed authors, they should be included.